airflow/www/views.py

```diff
@@ -1654,30 +1655,37 @@ def index(self):
         session.expunge_all()
         session.commit()
         session.close()
-        dags = dagbag.dags.values()
-        if do_filter:
-            if owner_mode == 'ldapgroup':
-                dags = {
-                    dag.dag_id: dag
-                    for dag in dags
-                    if (
-                        dag.owner in current_user.ldap_groups and (not dag.parent_dag)
-                    )
-                }
-            elif owner_mode == 'user':
-                dags = {
-                    dag.dag_id: dag
-                    for dag in dags
-                    if (
-                        dag.owner == current_user.user.username and (not dag.parent_dag)
-                    )
-                }
+
+        # read webserver_dags from local dagbag
+
+        webserver_dags = dagbag.dags.values()
+        # remove subdags
+        webserver_dags = [dag for dag in webserver_dags if not dag.parent_dag]
+
+        if do_filter and owner_mode == 'ldapgroup':
+            # only show dags owned by someone in @current_user.ldap_groups
+            webserver_dags = {
+                dag.dag_id: dag
+                for dag in webserver_dags
+                if dag.owner in current_user.ldap_groups
+            }
+        elif do_filter and owner_mode == 'user':
+            # only show dags owned by @current_user.user.username
+            webserver_dags = {
+                dag.dag_id: dag
+                for dag in webserver_dags
+                if dag.owner == current_user.user.username
+            }
         else:
-            dags = {dag.dag_id: dag for dag in dags if not dag.parent_dag}
-        all_dag_ids = sorted(set(orm_dags.keys()) | set(dags.keys()))
+            webserver_dags = {
```

Comment:
you can pull this out of and before the if statement, also this should be webserver_dag_ids (otherwise it's confusing to use the same variable to store two different types of data).